src/hotspot/share/runtime/mutexLocker.cpp

```diff
@@ -372,3 +372,26 @@ void print_owned_locks_on_error(outputStream* st) {
   }
   if (none) st->print_cr("None");
 }
+
+void unlock_locks_owned_by(Thread* thread) {
+  assert(thread != NULL, "can't be owned by NULL");
+  if (thread->is_Watcher_thread()) {
+    // need WatcherThread as a safeguard against potential deadlocks
+    return;
+  }
+
+#ifdef ASSERT
+  Mutex* owned_lock = thread->owned_locks();
+  while (owned_lock != NULL) {
+    Mutex* next = owned_lock->next();
+    owned_lock->unlock();
+    owned_lock = next;
+  }
+#endif // ASSERT
```

Comment:
Should we also clear `_owned_locks` after this so that it is still correct?

Reply:
In debug build, Thread keeps tracks its owning mutexes and Mutex keeps tracks its owner. 
eg. Mutex::unlock() -> Mutex::set_owner(NULL) -> Mutex::set_owner_implementation(NULL). 

here it deletes owned_lock from its owner.  
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/mutex.cpp#L483

Reply:
Thanks - I missed that housekeeping.